packages/input/src/input.tsx

```diff
@@ -47,15 +64,23 @@ export interface InputProps
  * Element that allows users enter single valued data.
  */
 export const Input = forwardRef<InputProps, "input">((props, ref) => {
-  const styles = useMultiStyleConfig("Input", props)
-  const ownProps = omitThemingProps(props)
+  const { htmlSize, ...rest } = props
+
+  const styles = useMultiStyleConfig("Input", rest)
+  const ownProps = omitThemingProps(rest)
   const input = useFormControl<HTMLInputElement>(ownProps)
   const _className = cx("chakra-input", props.className)
 
+  // This is used to remove the default width=100% when size is set, which otherwise conflicts
+  const { width, ...fieldStyles } = styles.field
+  const cssStyles = htmlSize ? { ...fieldStyles } : { ...styles.field }
+
   return (
     <chakra.input
+      as={NativeInput}
+      htmlSize={htmlSize}
       {...input}
-      __css={styles.field}
+      __css={{ ...cssStyles }}
```

Comment:
We can safely omit the spread operator here: 

```suggestion
      __css={cssStyles}
```

Reply:
Right that's a relict from testing. Remove it.